lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -348,20 +329,16 @@ public function setDirty($dirty)
      * Sets the initialized flag of the collection, forcing it into that state.
      *
      * @param bool $bool
-     *
-     * @return void
      */
-    public function setInitialized($bool)
+    public function setInitialized($bool): void
     {
         $this->initialized = $bool;
     }
 
     /**
      * {@inheritdoc}
-     *
-     * @return object
      */
-    public function remove($key)
+    public function remove($key): object
```

Comment:
should be `?object`. It calls parent::remove, ArrayCollection returns null in some cases and interface also says T|null.

please also take care of references to this method, like offsetUnset